app/src/test/java/org/mozilla/fenix/utils/BrowsersCacheTest.kt

```diff
@@ -0,0 +1,146 @@
+    // the application we are testing is recognized as a browser itself!
+
+    @Test
+    fun testInstallNewBrowsersWithoutReset() {
+        pretendBrowsersAreInstalled(
+            browsers = listOf(
+                Browsers.KnownBrowser.FIREFOX_NIGHTLY.packageName,
+                Browsers.KnownBrowser.REFERENCE_BROWSER.packageName
+            )
+        )
+
+        var browsers = BrowsersCache.all(testContext)
+        assertEquals(browsers.installedBrowsers.size, 3)
+
+        pretendBrowsersAreInstalled(
+            browsers = listOf(
+                Browsers.KnownBrowser.FIREFOX_NIGHTLY.packageName,
+                Browsers.KnownBrowser.FIREFOX.packageName,
+                Browsers.KnownBrowser.CHROME.packageName,
+                Browsers.KnownBrowser.SAMSUNG_INTERNET.packageName,
+                Browsers.KnownBrowser.DUCKDUCKGO.packageName,
+                Browsers.KnownBrowser.REFERENCE_BROWSER.packageName
+            )
+        )
+        browsers = BrowsersCache.all(testContext)
+        assertEquals(browsers.installedBrowsers.size, 3)
+    }
+
+    @Test
+    fun testInstallNewBrowsersWithReset() {
+        pretendBrowsersAreInstalled(
+            browsers = listOf(
+                Browsers.KnownBrowser.FIREFOX_NIGHTLY.packageName,
+                Browsers.KnownBrowser.REFERENCE_BROWSER.packageName
+            )
+        )
+
+        var browsers = BrowsersCache.all(testContext)
+        assertEquals(browsers.installedBrowsers.size, 3)
+
+        pretendBrowsersAreInstalled(
+            browsers = listOf(
+                Browsers.KnownBrowser.FIREFOX_NIGHTLY.packageName,
+                Browsers.KnownBrowser.FIREFOX.packageName,
+                Browsers.KnownBrowser.CHROME.packageName,
+                Browsers.KnownBrowser.SAMSUNG_INTERNET.packageName,
+                Browsers.KnownBrowser.DUCKDUCKGO.packageName,
+                Browsers.KnownBrowser.REFERENCE_BROWSER.packageName
+            )
+        )
+
+        BrowsersCache.resetAll()
+
+        browsers = BrowsersCache.all(testContext)
+        assertEquals(browsers.installedBrowsers.size, 7)
+    }
+
+    @Test
+    fun testResetAll() {
+        BrowsersCache.all(testContext)
+
+        assertNotNull(BrowsersCache.cachedBrowsers)
+        assertNotNull(BrowsersCache.cachedContext)
+
+        BrowsersCache.resetAll()
+
+        assertNull(BrowsersCache.cachedBrowsers)
+        assertNull(BrowsersCache.cachedContext)
+    }
+
+    private fun pretendBrowsersAreInstalled(
+        browsers: List<String> = listOf(),
+        defaultBrowser: String? = null,
+        url: String = "http://www.mozilla.org",
+        browsersExported: Boolean = true,
+        defaultBrowserExported: Boolean = true
+    ) {
+        val packageManager = testContext.packageManager
+        val shadow = shadowOf(packageManager)
+
+        browsers.forEach { packageName ->
+            val intent = Intent(Intent.ACTION_VIEW)
```

Comment:
nit: idiomatic kotlin is:
```kotlin
val intent = Intent(Intent.ACTION_VIEW).apply {
    package = packageName
    data = Uri.parse(url)
}
```

Here and below.

Reply:
Again, this was not my code but I am happy to make that change.

Reply:
Since it's not your code, no need to clean it up but see my thoughts above.